### PR TITLE
Weekly top DEV articles

### PR DESCRIPTION
**Pull request by king-11**
- nodemon setup
- article module
- lucy-bot endpoint update

### CODE REVIEW

File: .github/workflows/cron.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+on:
+  schedule:
+    - cron: "* * * * 0"
+
+jobs:
+  ping_bot:
+    runs-on: ubuntu-latest
+
+    name: Lucy Fetch Article on Ping
```
Comment (nishantwrp) [inline .github/workflows/cron.yml]: Maybe change it to `Weekly Top DEV Articles`

File: .env.example [hunk 2]
```diff
@@ -1 +0,0 @@
-LUCY_BOT_TOKEN=
```
Comment (nishantwrp) [inline .env.example:1]: Why remove this?

File: api/article.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import axios from 'axios';
+import { ExtraEditMessage } from 'telegraf/typings/telegram-types';
+import { ContextMessageUpdate, Extra } from 'telegraf';
+interface Article {
```
Comment (nishantwrp) [inline api/article.ts:18]: new line

File: api/lucy-bot.ts [hunk 4]
```diff
@@ -48,7 +45,18 @@ if (!PROD_ENV) {
...
+    await bot.telegram
+      .sendMessage(
+        `${process.env.CHATID}`,
+        `Hey Devs here are week's top blogs :)\n\n${replyList.join('\n\n')}`,
+        <ExtraEditMessage>Extra.markdown().webPreview(false),
+      )
+      .catch((e) => console.error(e));
+
+    resp.status(200).send('Article Sent');
+  }
```
Comment (nishantwrp) [inline api/lucy-bot.ts]: Are changes in this file intentional?

File: nodemon.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    "api/",
+    "./bot.ts"
+  ],
+  "ext": "js, ts, json",
+  "env": {
+    "LUCY_BOT_TOKEN": "your",
+    "URL": "your",
+    "CHATID": "your"
+  }
+}
```
Comment (nishantwrp) [inline nodemon.json]: newline

File: package.json [hunk 6]
```diff
@@ -11,7 +11,7 @@
   "main": "lib/index.js",
   "dependencies": {
     "@octokit/rest": "^18.0.6",
-    "dotenv": "^8.2.0",
+    "axios": "^0.20.0",
```
Comment (nishantwrp) [inline package.json:14]: why remove `dotenv`?

File: package.json [hunk 7]
```diff
@@ -27,4 +27,4 @@
     "typescript": "^3.2.2"
   },
   "private": true
-}
+}
```
Comment (nishantwrp) [inline package.json]: new line

**Review by nishantwrp — CHANGES_REQUESTED** [review]
Looks good generally. Some comments ptal!

File: api/lucy-bot.ts (hunk 4 above)
Comment (king-11) [reply api/lucy-bot.ts]: @nishantwrp I wasn't able to export the bot properly, the bot has to be there to send a message so I used the request, response of lucy-boy.ts which can be changed to something like a query param check to ensure that no one gets to send a message just on pinging lucy's link although I believe the link is already hidden

Comment (nishantwrp) [reply api/lucy-bot.ts]: Aren't the articles handled by the `article.ts` file?

File: api/lucy-bot.ts [hunk 8]
```diff
@@ -48,7 +44,7 @@ if (!PROD_ENV) {
   bot.launch();
 }
 
-module.exports = (req: NowRequest, resp: NowResponse) => {
+module.exports = async (req: NowRequest, resp: NowResponse): Promise<NowResponse> => {
   if (req.method === 'POST') bot.handleUpdate(req.body, resp);
-  else resp.status(200).send('Use POST to use Telegram bot!');
+  return resp.status(200).send('Use POST to use Telegram bot!');
```
Comment (nishantwrp) [inline api/lucy-bot.ts]: Is this change required?

File: api/lucy-bot.ts [hunk 9]
```diff
@@ -5,13 +5,9 @@ import { ExtraEditMessage } from 'telegraf/typings/telegram-types';
...
 const PROD_ENV = process.env.NODE_ENV === 'production';
 
-if (!PROD_ENV) {
-  // eslint-disable-next-line @typescript-eslint/no-var-requires
-  require('dotenv').config();
-}
-
 const bot = new Telegraf(process.env.LUCY_BOT_TOKEN || '');
 
+bot.use(Telegraf.log());
```
Comment (nishantwrp) [inline api/lucy-bot.ts:10]: What does `Telegraph.log()` do exactly?

File: api/article.ts [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+bot.use(Telegraf.log());
+bot.use(async (ctx: ContextMessageUpdate, next) => {
+  const start = new Date();
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  await next();
+  const ms = new Date().getTime() - start.getTime();
+  console.log('Response time: %sms', ms);
+});
+interface Article {
```
Comment (nishantwrp) [inline api/article.ts:18]: add a new line here

File: nodemon.json [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    "ts": "ts-node"
+  },
+  "watch": [
+    "api/"
+  ],
+  "delay": "2000",
+  "ext": "js,ts,json",
+  "env": {
+    "LUCY_BOT_TOKEN": "set",
+    "URL": "set",
```
Comment (nishantwrp) [inline nodemon.json]: Is this required?

**Review by nishantwrp — COMMENTED** [review]
Few questions, ptal!

File: api/lucy-bot.ts (hunk 9 above)
Comment (king-11) [reply api/lucy-bot.ts:10]: It will provide a detailed log something like this which can be useful to locate issue
```json
{
  "update_id": 24312332,
  "message": {
    "message_id": 173,
    "from": {
      "id": 862301703,
      "is_bot": false,
      "first_name": "Target_X",
      "username": "Target_X",
      "language_code": "en"
    },
    "chat": {
      "id": -450132051,
      "title": "test",
      "type": "group",
      "all_members_are_administrators": true
    },
    "date": 1602702162,
    "text": "/DevTalks",
    "entities": [
      {
        "offset": 0,
        "length": 9,
        "type": "bot_command"
      }
    ]
  }
}
```

File: api/lucy-bot.ts (hunk 8 above)
Comment (king-11) [reply api/lucy-bot.ts]: just additional type checking can be removed

File: api/article.ts (hunk 10 above)
Comment (king-11) [reply api/article.ts:18]: done

File: api/lucy-bot.ts (hunk 9 above)
Comment (nishantwrp) [reply api/lucy-bot.ts:10]: Oh, okay. In vercel logs?

File: nodemon.json (hunk 11 above)
Comment (king-11) [reply nodemon.json]: URL isn't needed rest are needed for proper working of article endpoint

File: api/lucy-bot.ts (hunk 8 above)
Comment (nishantwrp) [reply api/lucy-bot.ts]: Okay. then

File: nodemon.json (hunk 11 above)
Comment (nishantwrp) [reply nodemon.json]: Yup, remove it then

File: api/lucy-bot.ts (hunk 9 above)
Comment (king-11) [reply api/lucy-bot.ts:10]: yeah

Comment (nishantwrp) [reply api/lucy-bot.ts:10]: Cool!

File: nodemon.json (hunk 11 above)
Comment (king-11) [reply nodemon.json]: okay

File: api/lucy-bot.ts (hunk 8 above)
Comment (nishantwrp) [reply api/lucy-bot.ts]: Not resolved?

File: .github/workflows/cron.yml [hunk 12]
```diff
@@ -0,0 +1,22 @@
+name: Weekly Cron
+
+on:
+  schedule:
+    - cron: "11 11 * * 4"
```
Comment (nishantwrp) [inline .github/workflows/cron.yml]: oh and this was `* * * * 0` earlier? Is this correct?

Comment (king-11) [reply .github/workflows/cron.yml]: It was wrong cron syntax "* * * * 0" was every minute on sunday keep on pinging lucy now its for Thursday 11:11

**Review by king-11 — COMMENTED** [review]
All changes made

**Review by nishantwrp — APPROVED** [review]
lgtm!
